### ISSUE TITLE
pip install fail on both anaconda 3.5 and 3.6 OSX 10.11

### ISSUE THREAD

**Issue by melMass** [cprogrammer1994/ModernGL#138]
### Expected behavior

`pip install ModernGL`

### Actual behavior

Fail with exit status 1

### Steps to reproduce behavior

`pip install ModernGL`

### Version info

```
Any see output
```

```
Collecting ModernGL
  Using cached ModernGL-4.1.12.tar.gz
Building wheels for collected packages: ModernGL
  Running setup.py bdist_wheel for ModernGL ... error
  Complete output from command /Users/imac/anaconda/envs/SysCon/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-build-aen9evas/ModernGL/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/tmpq4_ohre0pip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.6
  creating build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/__main__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/buffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/common.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/context.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/framebuffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/members.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/mock.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/programs.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/renderbuffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/textures.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  copying ModernGL/vertex_arrays.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
  creating build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
  copying ModernGL/ext/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
  copying ModernGL/ext/__main__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
  running build_ext
  building 'ModernGL.mgl' extension
  creating build/temp.macosx-10.9-x86_64-3.6
  creating build/temp.macosx-10.9-x86_64-3.6/src
  /usr/local/bin/gcc-7 -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/imac/anaconda/envs/SysCon/include -arch x86_64 -I/Users/imac/anaconda/envs/SysCon/include -arch x86_64 -Isrc -I/Users/imac/anaconda/envs/SysCon/include/python3.6m -c src/Attribute.cpp -o build/temp.macosx-10.9-x86_64-3.6/src/Attribute.o -Wno-deprecated-declarations
  cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
  In file included from src/Attribute.cpp:3:0:
  src/Error.hpp:6:2: error: 'PyException_HEAD' does not name a type; did you mean 'PyObject_HEAD'?
    PyException_HEAD
    ^~~~~~~~~~~~~~~~
    PyObject_HEAD
  src/Attribute.cpp: In function 'PyObject* MGLAttribute_get_shape(MGLAttribute*, void*)':
  src/Attribute.cpp:55:9: error: 'PyUnicode_FromFormat' was not declared in this scope
    return PyUnicode_FromFormat("%c", self->shape);
           ^~~~~~~~~~~~~~~~~~~~
  src/Attribute.cpp:55:9: note: suggested alternative: 'PyUnicode_Format'
    return PyUnicode_FromFormat("%c", self->shape);
           ^~~~~~~~~~~~~~~~~~~~
           PyUnicode_Format
  error: command '/usr/local/bin/gcc-7' failed with exit status 1

  ----------------------------------------
  Failed building wheel for ModernGL
  Running setup.py clean for ModernGL
Failed to build ModernGL
Installing collected packages: ModernGL
  Running setup.py install for ModernGL ... error
    Complete output from command /Users/imac/anaconda/envs/SysCon/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-build-aen9evas/ModernGL/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-c_1nmlyi-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.6
    creating build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/__main__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/buffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/common.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/context.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/framebuffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/members.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/mock.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/programs.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/renderbuffers.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/textures.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    copying ModernGL/vertex_arrays.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL
    creating build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
    copying ModernGL/ext/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
    copying ModernGL/ext/__main__.py -> build/lib.macosx-10.9-x86_64-3.6/ModernGL/ext
    running build_ext
    building 'ModernGL.mgl' extension
    creating build/temp.macosx-10.9-x86_64-3.6
    creating build/temp.macosx-10.9-x86_64-3.6/src
    /usr/local/bin/gcc-7 -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/imac/anaconda/envs/SysCon/include -arch x86_64 -I/Users/imac/anaconda/envs/SysCon/include -arch x86_64 -Isrc -I/Users/imac/anaconda/envs/SysCon/include/python3.6m -c src/Attribute.cpp -o build/temp.macosx-10.9-x86_64-3.6/src/Attribute.o -Wno-deprecated-declarations
    cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
    In file included from src/Attribute.cpp:3:0:
    src/Error.hpp:6:2: error: 'PyException_HEAD' does not name a type; did you mean 'PyObject_HEAD'?
      PyException_HEAD
      ^~~~~~~~~~~~~~~~
      PyObject_HEAD
    src/Attribute.cpp: In function 'PyObject* MGLAttribute_get_shape(MGLAttribute*, void*)':
    src/Attribute.cpp:55:9: error: 'PyUnicode_FromFormat' was not declared in this scope
      return PyUnicode_FromFormat("%c", self->shape);
             ^~~~~~~~~~~~~~~~~~~~
    src/Attribute.cpp:55:9: note: suggested alternative: 'PyUnicode_Format'
      return PyUnicode_FromFormat("%c", self->shape);
             ^~~~~~~~~~~~~~~~~~~~
             PyUnicode_Format
    error: command '/usr/local/bin/gcc-7' failed with exit status 1

    ----------------------------------------
Command "/Users/imac/anaconda/envs/SysCon/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-build-aen9evas/ModernGL/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-c_1nmlyi-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /private/var/folders/z9/h5x_crxx3j3_6nry8w2jcwbh0000gn/T/pip-build-aen9evas/ModernGL/
```

**Comment by cprogrammer1994** [comment]
Thank you for the feedback!
I will investigate this issue.

[PyUnicode_FromFormat](https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromFormat) is not defined in you python header files.

I will search for alternatives/soulution for anaconda.

**Comment by melMass** [comment]
Thanks I'll investigate on my side as well

**Comment by cprogrammer1994** [comment]
can you please upload the include folder from anaconda?

My mac build server is broken and I am not able to test this

**Comment by melMass** [comment]
Sure can.
Do you have a preferred way for sending those ?

**Comment by melMass** [comment]
Here you go:

https://files.fm/u/z7bbuugg

Contains Headers of 3.5 and 3.6

Thanks

**Comment by cprogrammer1994** [comment]
Thanks! I will compare these files.

I took a quick look and I found the following:

In your **unicodeobject.h** PyUnicode_FromFormat can be found.

```c++
PyAPI_FUNC(PyObject *) PyUnicode_FromFormat(
    const char *format,   /* ASCII-encoded string  */
    ...
    );
```

**Comment by cprogrammer1994** [comment]
Sorry for this, but I cannot reproduce the problem. my mac build-system is down. I tested on windows. ModernGL seems to install fine. I also uploaded a conda package. I hope this helps.

**Comment by cprogrammer1994** [comment]
Wheel files are available for Python 3.4, Python 3.5 and Python 3.6 on Mac. It should work in your case too. Closing.
